nx/lib/nx/lin_alg/svd.ex

```diff
@@ -24,16 +24,53 @@ defmodule Nx.LinAlg.SVD do
   defn svd(input_tensor, opts \\ []) do
     validate_opts(opts)
 
-    result =
-      if Nx.all(input_tensor == 0) do
-        svd_all_zeros(input_tensor, opts)
-      else
-        svd_non_zero(input_tensor, opts)
-      end
+    input_tensor = collapse_batch_axes_and_vectorize(input_tensor)
+
+    {uz, sz, vtz} = svd_all_zeros(input_tensor, opts)
+    {u, s, vt} = svd_non_zero(input_tensor, opts)
+
+    all_zeros = Nx.all(input_tensor == 0)
+
+    u = Nx.select(all_zeros, uz, u)
+    s = Nx.select(all_zeros, sz, s)
+    vt = Nx.select(all_zeros, vtz, vt)
+
+    result = {u, s, vt}
+
+    {u, s, vt} =
+      custom_grad(result, [input_tensor], fn g ->
+        svd_grad(result, input_tensor, g)
+      end)
+
+    u = Nx.devectorize(u, keep_names: false)
+    s = Nx.devectorize(s, keep_names: false)
+    vt = Nx.devectorize(vt, keep_names: false)
```

Comment:
Somehow each time we put vectorized values into a tuple and then extract them back, the rightmost value is coming back devectorized with `keep_names: true`, so these devectorize calls won't work correctly

Reply:
After we fix this I'll add more tests because we still need to "uncollapse" batch axes after we devectorize here